MoviesTVSentiments/app/src/main/java/com/google/moviestvsentiments/service/database/AsyncDatabaseExecutor.java

```diff
@@ -0,0 +1,42 @@
+package com.google.moviestvsentiments.service.database;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * A DatabaseExecutor that executes all queries on background threads.
+ */
+public class AsyncDatabaseExecutor implements DatabaseExecutor {
```

Comment:
Maybe you can take a look at ThreadPoolExecutor and see if that is more suitable: https://developer.android.com/reference/java/util/concurrent/ThreadPoolExecutor.

It is totally fine if you still decide to use Executors factory methods, but consider specifying a ThreadFactory so that the database thread runs at a lower priority than UI thread.

Reply:
I used newCachedThreadPool, but lowered the priority of the threads.